src/mqt/predictor/utils.py

```diff
@@ -251,45 +199,44 @@ def calc_eval_score_for_qc(qc_path: str, device: str):
                         )
                     )
                     if (
-                        rigetti_m1_calibration["fid_2Q_CZ"].get(tmp) is None
-                        or rigetti_m1_calibration["fid_2Q_CZ"][tmp] is None
+                        rigetti_m2_calibration["fid_2Q_CZ"].get(tmp) is None
+                        or rigetti_m2_calibration["fid_2Q_CZ"][tmp] is None
                     ):
-                        specific_fidelity = rigetti_m1_calibration["avg_2Q"]
+                        specific_fidelity = rigetti_m2_calibration["avg_2Q"]
                     else:
-                        specific_fidelity = rigetti_m1_calibration["fid_2Q_CZ"][tmp]
+                        specific_fidelity = rigetti_m2_calibration["fid_2Q_CZ"][tmp]
 
                 res *= specific_fidelity
 
-    elif "ionq11" in device:
-        for instruction, qargs, _cargs in qc.data:
-            gate_type = instruction.name
-            qubit_indices = [elem.index for elem in qargs]
-
-            assert gate_type in ["rxx", "rz", "ry", "rx", "measure", "barrier"]
-            if gate_type != "barrier":
-                assert len(qubit_indices) in [1, 2]
-
-                if len(qubit_indices) == 1:
-                    specific_fidelity = ionq_calibration["avg_1Q"]
-                elif len(qubit_indices) == 2:
-                    specific_fidelity = ionq_calibration["avg_2Q"]
-                res *= specific_fidelity
     else:
         print("Error: No suitable backend found!")
 
     return res
 
 
+def calc_qubit_index(qargs, qregs, index):
+    offset = 0
+    for reg in qregs:
+        if qargs[index] not in reg:
+            offset += reg.size
+        else:
+            qubit_index = offset + reg.index(qargs[index])
+            return qubit_index
+    raise ValueError("Qubit not found.")
+
+
 def init_all_config_files():
     try:
+        global ibm_washington_cx_mean_error
+        ibm_washington_cx_mean_error = get_mean_IBM_washington_cx_error()
         global ibm_montreal_calibration
         ibm_montreal_calibration = FakeMontreal().properties()
         global ibm_washington_calibration
         ibm_washington_calibration = FakeWashington().properties()
         global oqc_lucy_calibration
         oqc_lucy_calibration = parse_oqc_calibration_config()
-        global rigetti_m1_calibration
-        rigetti_m1_calibration = parse_rigetti_calibration_config()
+        global rigetti_m2_calibration
+        rigetti_m2_calibration = parse_rigetti_calibration_config()
         global ionq_calibration
         ionq_calibration = parse_ionq_calibration_config()
```

Comment:
this also looks like the perfect use-case for a dedicated class in a dedicated file that encapsulates this calibration data and its setup.

Reply:
I commited a new structured. Have you meant it that way or do you have further suggestions?